### PR TITLE
WT-11535 Create a dedicated file truncate function for chunkcache

### PR DESCRIPTION
**Pull request by Siddhartha8899**
This PR was created to deal with a recent failure in a chunkcache test on `Linux noftruncate`  build variant.

To enable the successful operation of the chunkcache on `Linux noftruncate`  build variants, the following has been done:

- A function for truncating the chunkcache file.
- Within this function, an attempt is made to truncate the file, depending on its support.
- In cases where support is absent, a manual process of resetting the file to a zeroed state is executed.

There are some modifications made to the existing function `__wt_file_zero`. @sueloverso it would be great if you could review the same.

### CODE REVIEW

File: src/log/log.c
Comment (sueloverso) [inline src/log/log.c:1416]: Sending in a pointer is awkward here. I see what you're doing and why. @ddanderson you might have some thoughts here too.
1. Should the chunkcache be subject to IO capacity and have its own type? It has its own storage path but that may be on the same IO channel. The answer to that isn't clear and bigger than this ticket. The read/write paths in the chunkcache should also be subject to throttling if yes. My answer is that yes, it should be subject to throttling if it uses the same channel. But no if it storage_path is not the same channel. I don't know if we can tell. @fedorova any insight?
3. For this usage, I'd prefer you add in a WT_THROTTLE_UNKNOWN or something and send in a value rather than the complicated pointer usage here.

**Review by sueloverso — COMMENTED** [review]
I have one larger comment here that I'd like others opinions on before this is merged.

File: src/log/log.c
Comment (Siddhartha8899) [reply src/log/log.c:1416]: Thanks for the feedback, @sueloverso. I'm open to removing the pointer usage. However, I'll hold off on making that change until we have a resolution on [WT-11537](https://jira.mongodb.org/browse/WT-11537), especially given the recent queries about supporting the Linux Ftruncate build variant.

As for your initial observation, as you have already mentioned it's beyond the scope of this ticket as the primary aim here is to address the CI blocker, but I'm definitely open to discussing it further in this thread and if required, happy to create its own dedicated ticket.

Comment (ddanderson) [reply src/log/log.c:1416]: I agree with @sueloverso that we should choose a real value to throttle.  I would suggest: create a `WT_THROTTLE_CHUNK_CACHE` enum, and using it in place of the current NULL usage.  Then in `__wt_capacity_throttle` in the first switch statement, add:
```
   case WT_THROTTLE_CHUNK_CACHE:
     /* At the moment, chunk cache usages are not throttled. */
     return;
```
There's another switch statement further down, you should probably add:
```
   case WT_THROTTLE_CHUNK_CACHE:
        return (__wt_illegal_value(session, type));
```
I think this is groundwork we need no matter what the resolution of WT-11537, because even if we don't need to throttle an `ftruncate` call in the chunk cache code, we may need to throttle in other places in the chunk cache.  I'm going to open another ticket to track that.

Comment (ddanderson) [reply src/log/log.c:1416]: The other ticket is https://jira.mongodb.org/browse/WT-11542

Comment (Siddhartha8899) [reply src/log/log.c]: Thank you @ddanderson for the review and for creating the ticket. 

Updated as discussed in [72ee2a8](https://github.com/wiredtiger/wiredtiger/pull/9549/commits/72ee2a84609ae22ce273c0ee036cda0c47886ad2).

For the time being, I've left `calling __wt_file_zero` in this PR. If we eventually opt to eliminate the `Linux NoFtruncate build variant` (in [WT-11537](https://jira.mongodb.org/browse/WT-11537)), we can subsequently remove it from this location. Presumably this removal process would need to be done in other instances.

**Review by korteland — APPROVED** [review]
Thanks - this looks good to me.

**Pull request merged**
